### PR TITLE
[INLONG-5205][Sort] Add reporting metrics for Elasticsearch 6 connector

### LINKED ISSUE #5205: [Feature][Sort] Add reporting metrics for Elasticsearch 6 connector
### Description

Add reporting metrics for Elasticsearch 6 connector

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Oneal65**
- Fixes #5205 

### Motivation

Add reporting metrics for Elasticsearch 6 connector

### Modifications

Add reporting metrics for Elasticsearch 6 connector

### some reference

there is a pr in flink-connector: https://github.com/apache/flink/pull/8568

### Verifying this change

*(Please pick either of the following options)*

- [x] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-sort/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/ElasticsearchSinkBase.java [hunk 1]
```diff
@@ -263,8 +265,23 @@ public void disableFlushOnCheckpoint() {
     @Override
     public void open(Configuration parameters) throws Exception {
         client = callBridge.createClient(userConfig);
+        metricData = new MetricData(getRuntimeContext().getMetricGroup());
```
Comment (gong) [inline inlong-sort/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/ElasticsearchSinkBase.java:268]: metricData can be replace by class of connector-base

**Review by healchow — APPROVED** [review]

**Review by yunqingmoswu — APPROVED** [review]
LGTM

File: inlong-sort/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/table/ElasticsearchOptions.java [hunk 2]
```diff
@@ -155,8 +144,23 @@ public enum BackOffType {
                     .withDescription(
                             "The format must produce a valid JSON document. "
                                     + "Please refer to the documentation on formats for more details.");
+    public static final ConfigOption<String> INLONG_METRIC =
```
Comment (EMsnap) [inline inlong-sort/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/table/ElasticsearchOptions.java:147]: inlong metric is defined in inlongMetric interface, can just use it

Comment (Oneal65) [reply inlong-sort/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/table/ElasticsearchOptions.java]: DONE

**Review by dockerzhang — APPROVED** [review]

**Review by EMsnap — APPROVED** [review]

**Pull request merged**